### PR TITLE
Tweak BuilderInfo and FieldInfo docs:

### PR DESCRIPTION
**Pull request by osa1**
- Fix documentation syntax, add references to other classes, properties,
  methods

- Document undocumented properties

- Remove references to non-existent identifiers

- Some rewording

### CODE REVIEW

File: protobuf/lib/src/protobuf/field_info.dart [hunk 1]
```diff
@@ -13,39 +13,49 @@ class FieldInfo<T> {
   /// This will typically be in camel case.
   final String name;
 
-  /// The name of this field as written in the proto-definition.
+  /// Name of this field as written in the proto definition.
```
Comment (sigurdm) [inline protobuf/lib/src/protobuf/field_info.dart:16]: Would be nice with an example also here and above :)

**Review by sigurdm — APPROVED** [review]
LGTM

**Comment by osa1** [comment]
@sigurdm addressed your comments. It would be great if you could proof read 😄

**Review by sigurdm — APPROVED** [review]
LGTM

**Pull request merged**